materialize-sql/std_sql.go

```diff
@@ -46,20 +46,23 @@ func StdSQLExecStatements(ctx context.Context, db *sql.DB, statements []string)
 	// through a single connection. This allows a driver to explicitly run
 	// `BEGIN;` and `COMMIT;` statements around a transactional operation.
 	var conn, err = db.Conn(ctx)
-	if err == nil {
```

Comment:
Do we still need these changes? The changes in `materialize-databricks` generally make sense. This change seems unnecessary, since we we already doing `conn.Close()` (and actually returning errors from it) unless there was some other error that would crash the connector.

Reply:
@williamhbaker I think the best practice is to use `defer` in case of an early return to make sure we explicitly cleanup resources. That said, I found a way to return the error from `conn.Close` in defer so we can keep the best of both worlds.

Initially when I opened this pull-request I saw changes in how the warehouse waking up and the leaked connections behaved by addition of this, although I have a hard time reproducing that again... so I'd like to keep these.